packages/studio-be/src/core/users/workspace-service.ts

```diff
@@ -84,6 +84,24 @@ export class WorkspaceService {
     return list.find(x => x.workspace === workspace)
   }
 
+  async save(workspaces: Workspace[]): Promise<void> {
+    return this.ghost.global().upsertFile('/', 'workspaces.json', JSON.stringify(workspaces, undefined, 2))
+  }
+
+  async addBotRef(botId: string, workspaceId: string): Promise<void> {
+    const workspaces = await this.getWorkspaces()
+    const workspace = workspaces.find(x => x.id === workspaceId)
```

Comment:
I think you should move this piece of code into a function since it's copy-pasted at multiple occurrences in this file.

Reply:
Necessary here because we need to save all workspaces, not just the one we need. Else we have findWorkspace